apps/server/src/modules/lesson/uc/lesson.uc.spec.ts

```diff
@@ -68,4 +75,82 @@ describe('LessonUC', () => {
 
 		expect(result).toBe(true);
 	});
+
+	describe('getLesons', () => {
+		describe('when user is a valid teacher', () => {
+			const setup = () => {
+				const user = userFactory.buildWithId();
+				authorizationService.getUserWithPermissions.mockResolvedValueOnce(user);
+
+				const course = courseFactory.buildWithId();
+				courseService.findOneForUser.mockResolvedValueOnce(course);
+
+				const lesson = lessonFactory.buildWithId({ course });
+				const hiddenLesson = lessonFactory.buildWithId({ course, hidden: true });
+				lessonService.findByCourseIds.mockResolvedValueOnce([[lesson, hiddenLesson], 2]);
+				authorizationService.hasPermission.mockReturnValue(true);
+
+				return { user, course, lesson, hiddenLesson };
+			};
+			it('should get user with permissions from authrozationService', async () => {
+				const { user } = setup();
+				await lessonUC.getLessons(user.id, 'courseId');
+				expect(authorizationService.getUserWithPermissions).toHaveBeenCalledWith(user.id);
+			});
+			it('should get course from courseService', async () => {
+				const { user, course } = setup();
+				await lessonUC.getLessons(user.id, course.id);
+				expect(courseService.findOneForUser).toHaveBeenCalledWith(course.id, user.id);
+			});
+			it('should check user course permission', async () => {
+				const { user, course } = setup();
+				await lessonUC.getLessons(user.id, course.id);
+				expect(authorizationService.checkPermission).toHaveBeenCalledWith(
+					expect.objectContaining({ ...user }),
+					expect.objectContaining({ ...course }),
+					AuthorizationContextBuilder.read([Permission.COURSE_VIEW])
+				);
+			});
```

Comment:
the fact that we call the checkPermissions is sufficient I believe. That method will throw an unauthorized error if needed, and that part is not our business to test here.
Otherwise it would be like"when authorization throws an error, check that an error is thrown in our uc here", which is redundant. 
With the API tests I think we have pretty good coverage.